cts/cts-cli.in

```diff
@@ -2227,65 +2227,65 @@ class CrmShadowRegressionTest(RegressionTest):
+            Test("Reset active shadow instance to active CIB",
+                 "crm_shadow --reset {shadow} --batch"),
+            Test("Active shadow instance no different from active CIB after reset",
+                 "crm_shadow --diff"),
+            Test("Active shadow instance differs from active CIB after change",
+                 "crm_shadow --diff",
+                 setup="crm_attribute -n admin_epoch -v 99",
+                 expected_rc=ExitStatus.ERROR),
+
+            ValidatingTest("Reset active shadow instance to active CIB",
                            "crm_shadow --reset {shadow} --batch --output-as=xml"),
-            ValidatingTest("Get active shadow instance's diff (after reset)",
+            ValidatingTest("Active shadow instance no different from active CIB after reset",
                            "crm_shadow --diff --output-as=xml"),
-            # Reset an inactive shadow instance with no active instance
-            Test("Reset shadow instance (no active instance)",
-                 "crm_shadow --reset {shadow} --batch",
-                 env={"CIB_shadow": None}),
+            ValidatingTest("Active shadow instance differs from active CIB after change",
+                           "crm_shadow --diff --output-as=xml",
+                           setup="crm_attribute -n admin_epoch -v 199",
+                           expected_rc=ExitStatus.ERROR),
+
+            make_test_group("Reset shadow instance to active CIB with nonexistent shadow file",
+                            "crm_shadow --reset {shadow} --batch {fmt}",
+                            [Test, ValidatingTest],
+                            setup="crm_shadow --delete {shadow} --force",
+                            expected_rc=ExitStatus.NOSUCH),
+            make_test_group("Force-reset shadow instance to active CIB with nonexistent shadow file",
+                            "crm_shadow --reset {shadow} --batch --force {fmt}",
+                            [Test, ValidatingTest],
+                            setup="crm_shadow --delete {shadow} --force"),
+            Test("Active shadow instance no different from active CIB after force-reset",
+                 "crm_shadow --diff"),
         ]
 
         reset_2_tests = [
-            ValidatingTest("Reset shadow instance (no active instance)",
-                           "crm_shadow --reset {shadow} --batch --output-as=xml"),
-            # Reset an inactive shadow instance with an active instance
-            Test("Reset shadow instance (mismatch)",
-                 "crm_shadow --reset {shadow} --batch",
-                 env={"CIB_shadow": "nonexistent_shadow"},
-                 expected_rc=ExitStatus.USAGE),
-            Test("Reset shadow instance (mismatch) (force)",
-                 "crm_shadow --reset {shadow} --batch --force",
-                 env={"CIB_shadow": "nonexistent_shadow"}),
+            make_test_group("Reset inactive shadow instance (none active) to active CIB",
```

Comment:
> Previously, the shadow CIB was reset with text output then immediately
> reset with XML output. That's not a good test because there are no
> changes to reset in the XML case. Reset with text output, make a
> change, then reset with XML output.

I guess the above was referring only to the tests with `diff`. Here we don't make any change before resetting again with XML output here.

Reply:
Right, as long as we test it once, I'm not worried about these other cases